File: apps/ios/Sources/TePlannerKit/Services/LegalLinks.swift
import Foundation

/// Public URLs for the Tautomation legal documents. Used by the login
/// page consent line and the Settings → 关于 → 隐私政策 / 用户协议
/// navigation links.
///
/// Apple App Review requires these URLs to be live + reachable when
/// the binary is submitted — they're verified against the App Store
/// Connect metadata. Hosting plan documented in
/// `docs/legal/README.md`. If the URLs change, update both here AND
/// the ASC App Privacy / Support URL fields in lockstep, otherwise
/// review may reject the binary for an inconsistent privacy link.
public enum LegalLinks {

    /// Public privacy policy. Currently a placeholder — point at the
    /// gh-pages / api.teplanner.cloud/legal/ host once the static
    /// pages go live. The DRAFT markdown lives at
    /// `docs/legal/privacy-policy.md`.
    public static let privacyPolicy = URL(string: "https://teplanner.cloud/legal/privacy.html")!

    /// Public terms of service / user agreement.
    public static let termsOfService = URL(string: "https://teplanner.cloud/legal/terms.html")!
}

File: apps/android/app/src/main/java/cloud/teplanner/android/util/LegalLinks.kt
package cloud.teplanner.android.util

import android.content.Context
import android.content.Intent
import android.net.Uri

/**
 * Public URLs for the Tautomation legal documents. Used by the login
 * page consent line and the Settings → 关于 → 隐私政策 / 用户协议
 * row.
 *
 * Both URLs are placeholders until the markdown drafts under
 * `docs/legal/` are published to a public HTML host. They MUST be
 * live + reachable when the Play Store binary is submitted; reviewers
 * verify them against the listing metadata.
 *
 * Keep in lockstep with iOS [LegalLinks].
 */
object LegalLinks {
    const val PRIVACY_POLICY_URL = "https://teplanner.cloud/legal/privacy.html"
    const val TERMS_OF_SERVICE_URL = "https://teplanner.cloud/legal/terms.html"

    /** Open a URL in the system browser (Chrome / Huawei browser / etc.). */
    fun open(context: Context, url: String) {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        try {
            context.startActivity(intent)
        } catch (_: Throwable) {
            // No browser installed — extremely rare on Android. Silent
            // failure is acceptable; the user can also reach the URL
            // from any other device.
        }
    }
}
